examples/authResolvers.js

```diff
@@ -0,0 +1,84 @@
+const fs = require('fs')
+const path = require('path')
+const express = require('express')
+const session = require('express-session')
+const Keycloak = require('keycloak-connect')
+const { ApolloServer, gql } = require('apollo-server-express')
+
+const { KeycloakContextProvider, KeycloakTypeDefs, auth, hasRole } = require('../')
+
+const app = express()
+
+const memoryStore = new session.MemoryStore()
+
+const graphqlPath = '/graphql'
+
+
+app.use(session({
+  secret: 'secret',
+  resave: false,
+  saveUninitialized: true,
+  store: memoryStore
+}))
+
+const keycloakConfig = JSON.parse(fs.readFileSync(path.resolve(__dirname, './config/keycloak.json')))
+
+const keycloak = new Keycloak({
+  store: memoryStore
+}, keycloakConfig)
+
+// Install general keycloak middleware
+app.use(keycloak.middleware({
+  admin: graphqlPath
+}))
+
+// Protect the main route for all graphql services
+// Disable unauthenticated access
+app.use(graphqlPath, keycloak.middleware())
+
+const typeDefs = gql`
+  type Query {
+    greetings: [String]!
+  }
+
+  type Mutation {
+    addGreeting(greeting: String!): String!
+  }
+`
+
+const greetings = [
+  'hello world!'
+]
+
+const resolvers = {
+  Query: {
+    greetings: () => greetings
+  },
+  Mutation: {
+    addGreeting: auth(hasRole('developer')((obj, { greeting }, context, info) => {
```

Comment:
This looks strange but yeah. What is advantage of this api over the using it directly.in resolver

Reply:
Okay so I wouldn't actually recommend this approach but the benefit is that your authentication/authorization logic is as decoupled from your business logic as possible. If you read this [blog post on Authentication / Authorization in GraphQL](https://medium.com/the-guild/authentication-and-authorization-in-graphql-and-how-graphql-modules-can-help-fadc1ee5b0c2) this kind of approach is the second best, most recommended approach. Then the next level, number 1 approach is using `grahphql-modules`. 

The example in this case is just to demonstrate that those resolvers:

* will be flexible enough to be used in multiple environments. Example: these could be used in GraphQL Express where setting up directives might be a challenge or not necessary for your particular application.
* Are also structured in a way that they will be usable within GraphQL modules.